src/entities/fractions/fraction.rs

```diff
@@ -20,7 +20,15 @@ impl<M: Default> Default for FractionLike<M> {
     }
 }
 
-// Type alias for a simple Fraction without metadata
+impl Deref for FractionLike<Currency> {
+    type Target = Currency;
+
+    fn deref(&self) -> &Self::Target {
+        &self.meta
+    }
+}
```

Comment:
```suggestion
impl<M> Deref for FractionLike<M> {
    type Target = M;

    fn deref(&self) -> &Self::Target {
        &self.meta
    }
}
```
Do the same for `CurrencyLike<M>`. Refactor all occurrences of `.meta.`.